gammapy/estimators/flux_map.py

```diff
@@ -455,102 +470,119 @@ def eflux_ref(self):
-        return self.norm_ul * self.e2dnde_ref
+        return self._filter_convergence_failure(self.norm_ul * self.e2dnde_ref, status=2)
 
     @property
     def flux(self):
         """Return integral flux (flux) SED values."""
-        return self.norm * self.flux_ref
+        return self._filter_convergence_failure(self.norm * self.flux_ref)
 
     @property
     def flux_err(self):
         """Return integral flux (flux) SED values."""
-        return self.norm_err * self.flux_ref
+        return self._filter_convergence_failure(self.norm_err * self.flux_ref, status=2)
 
     @property
     def flux_errn(self):
         """Return integral flux (flux) SED negative errors."""
-        return self.norm_errn * self.flux_ref
+        return self._filter_convergence_failure(self.norm_errn * self.flux_ref, status=2)
 
     @property
     def flux_errp(self):
         """Return integral flux (flux) SED positive errors."""
-        return self.norm_errp * self.flux_ref
+        return self._filter_convergence_failure(self.norm_errp * self.flux_ref, status=2)
 
     @property
     def flux_ul(self):
         """Return integral flux (flux) SED upper limits."""
-        return self.norm_ul * self.flux_ref
+        return self._filter_convergence_failure(self.norm_ul * self.flux_ref, status=2)
 
     @property
     def eflux(self):
         """Return energy flux (eflux) SED values."""
-        return self.norm * self.eflux_ref
+        return self._filter_convergence_failure(self.norm * self.eflux_ref)
 
     @property
     def eflux_err(self):
         """Return energy flux (eflux) SED errors."""
-        return self.norm_err * self.eflux_ref
+        return self._filter_convergence_failure(self.norm_err * self.eflux_ref, status=2)
 
     @property
     def eflux_errn(self):
         """Return energy flux (eflux) SED negative errors."""
-        return self.norm_errn * self.eflux_ref
+        return self._filter_convergence_failure(self.norm_errn * self.eflux_ref, status=2)
 
     @property
     def eflux_errp(self):
         """Return energy flux (eflux) SED positive errors."""
-        return self.norm_errp * self.eflux_ref
+        return self._filter_convergence_failure(self.norm_errp * self.eflux_ref, status=2)
 
     @property
     def eflux_ul(self):
         """Return energy flux (eflux) SED upper limits."""
-        return self.norm_ul * self.eflux_ref
+        return self._filter_convergence_failure(self.norm_ul * self.eflux_ref, status=2)
+
+    @property
+    def fit_status(self):
+        """Fit statistic value"""
+        self._check_quantity("fit_status")
+        return self._data["fit_status"]
+
+    def _filter_convergence_failure(self, some_map, status=1):
+        """Put NaN where pixels did not converge."""
+        if not self.has_fit_status:
+            return some_map
+        # For integer content, NaN is not defined
+        if not np.asanyarray(some_map.data).dtype is np.dtype(float):
```

Comment:
This line is hard to understand. Is it useful? All fitted quantities are float no?

Also `Map` has a `dtype` attribute directly.

Reply:
There are not. A test function returns one integer, and not an ndarray of float. This is why I use np.asanyarray

Reply:
If it appears only in one test function, it would be simpler to adapt it rather than the base code